src/stream/src/executor/backfill/cdc/upstream_table/snapshot.rs

```diff
@@ -104,9 +107,28 @@ impl UpstreamTableRead for UpstreamTableReader<ExternalStorageTable> {
 
         let mut builder = DataChunkBuilder::new(self.inner.schema().data_types(), args.chunk_size);
         let chunk_stream = iter_chunks(row_stream, &mut builder);
+
+        if args.chunk_size == 0 {
+            // If limit is 0, we should not read any data from the upstream table.
+            // Keep waiting util the stream is rebuilt.
+            let future = futures::future::pending::<()>();
+            future.await;
+        }
+        let limiter = {
```

Comment:
@tabVersion does the rate limiter will always be created? IIRC, the default chunk_size is `256`, the snapshot stream will be limited to 256 rows/second, am I right?  If that's the case, we should fix it because it causes the snapshot become slow.

Reply:
hummmm I think yes, a bug indeed.

Reply:
Already fixed by https://github.com/risingwavelabs/risingwave/pull/16407.